enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/JWTAuthenticator.java

```diff
@@ -329,18 +329,31 @@ public AuthenticationContext authenticate(RequestContext requestContext) throws
                         requestContext.addMetadataToMap("ratelimit:usage-policy", subPolicyName);
 
                         String matchedApiOrganizationId = requestContext.getMatchedAPI().getOrganizationId();
-                        if (datastore.getSubscriptionPolicyByOrgIdAndName(matchedApiOrganizationId, subPolicyName)
-                                != null) {
-                            SubscriptionPolicy subPolicy = datastore.getSubscriptionPolicyByOrgIdAndName
-                                    (matchedApiOrganizationId, subPolicyName);
-                            String metaDataOrgId =
+                        // Denotes datastore contains a subscription policy for the given name and organization
+                        SubscriptionPolicy subPolicy = datastore.getSubscriptionPolicyByOrgIdAndName(
+                                matchedApiOrganizationId, subPolicyName);
+                        String metaDataOrgId = APIConstants.SUPER_TENANT_DOMAIN_NAME;
+                        if (subPolicy != null) {
+                            metaDataOrgId =
                                     FeatureFlags.getCustomSubscriptionPolicyHandlingOrg(subPolicy.getOrganization());
-                            log.debug("Subscription rate-limiting will be evaluated for the organization: " +
-                                    metaDataOrgId);
                             requestContext.addMetadataToMap("ratelimit:organization", metaDataOrgId);
+                            if (subPolicy.getRateLimitCount() > 0 &&
+                                    StringUtils.isNotEmpty(subPolicy.getRateLimitTimeUnit())) {
```

Comment:
If we check this condition in more places maybe we can move this condition to a function and reuse it.